docs/recipes/install/poc-markdown/common/parse_doc_md.py

```diff
@@ -0,0 +1,314 @@
+            pass
+
+        return output.getvalue()
+
+    def _process_line(self, line: str, output: StringIO) -> None:
+        """Process a single line."""
+        line = line.rstrip("\n\r")
+
+        # Handle ohpc_run blocks
+        if "<!-- begin_ohpc_run -->" in line:
+            self._in_ohpc_run = True
+            return
+        elif "<!-- end_ohpc_run -->" in line:
+            self._in_ohpc_run = False
+            self._in_code_block = False
+            return
+
+        if not self._in_ohpc_run:
+            return
+
+        # Handle code blocks
+        if match := self._PATTERNS["code_block_start"].match(line):
+            self._in_code_block = True
+            return
+        elif self._PATTERNS["code_block_end"].match(line):
+            self._in_code_block = False
+            return
+
+        # Process validation newline
+        if self._PATTERNS["validation_newline"].match(line.strip()):
+            output.write("\n")
+            return
+
+        # Process directives
+        if match := self._PATTERNS["ci_comment"].match(line.strip()):
+            if self.ci_run:
+                output.write(f"# {match.group(1)} (CI only)\n")
+        elif match := self._PATTERNS["header_comment"].match(line.strip()):
+            comment = self._process_refs(match.group(1))
+            self._write_header(output, comment)
+        elif match := self._PATTERNS["validation_comment"].match(line.strip()):
+            comment = self._process_refs(match.group(1))
+            output.write(f"# {comment}\n")
+        elif match := self._PATTERNS["command_directive"].match(line.strip()):
+            cmd = self._substitute_variables(match.group(1))
+            output.write(f"{cmd}\n")
+        elif self._in_code_block:
+            self._process_command_line(line, output)
+
+    def _process_command_line(self, line: str, output: StringIO) -> None:
+        """Process command lines in code blocks."""
+        line = line.strip()
+
+        # Skip empty lines and comments that aren't commands
+        if not line or (line.startswith("#") and not line.startswith("# ")):
+            return
+
+        # HERE document
+        if match := self._PATTERNS["sms_here_doc"].search(line):
+            self._process_here_doc(match, output)
+        # Line continuation
+        elif match := self._PATTERNS["sms_continuation"].search(line):
+            self._process_continuation(match, output)
+        # Do loop
+        elif match := self._PATTERNS["sms_do_loop"].search(line):
+            self._process_do_loop(match, output)
+        # Regular SMS prompt
+        elif match := self._PATTERNS["sms_prompt"].search(line):
+            cmd = self._substitute_variables(match.group(1))
+            output.write(f"{cmd}\n")
+        # Plain comments
+        elif line.startswith("#"):
+            comment = self._substitute_variables(line)
+            output.write(f"{comment}\n")
+
+    def _substitute_variables(self, cmd: str) -> str:
```

Comment:
Looks like this is unused and that the variable substitution is done by Jinja2?  Is this understanding correct?